exercises/collatz-conjecture/package.yaml

```diff
@@ -0,0 +1,21 @@
+
+name: collatz-conjecture
+version: 1.0.0.3
```

Comment:
as https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json has version 1.1.1, the version that would comply with our versioning policy in https://github.com/exercism/haskell/issues/522 is `1.1.1.1`, and ensuring that the tests in the Haskell test file match those in the JSON file

Reply:
(or, alternatively, if you believe those tests are unsuitable for the Haskell track for any reason,  let's discuss why they are unsuitable, and then use version`0.1.0.1`)